astropy/modeling/core.py

```diff
@@ -1098,6 +1219,15 @@ def _initialize_parameter_values(self, params):
 
             param_metrics[name]['slice'] = param_slice
             param_metrics[name]['shape'] = param_shape
+
+            if isinstance(value, Quantity):
```

Comment:
Note: We may be able to soften this to just allow anything with a `.unit` as in some of the other parts of Astropy. But for now I want to be stricter about this so that I can limit the number of assumptions that need to be made.